streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -1616,4 +1623,25 @@ public <VOut> KStream<K, VOut> processValues(
             processNode,
             builder);
     }
+
+    @Override
+    public KStream<K, V> markAsPartitioned() {
+        final ProcessorParameters<? super K, ? super V, ?, ?> processorParameters =
+                new ProcessorParameters<>(new PassThrough<>(), PARTITION_PRESERVE_NAME + name);
```

Comment:
We should not append `name` what is the name of the current `KStream`. Instead, we should generate a new local `name`:
```
final String name = builder.newProcessorName(PARTITION_PRESERVE_NAME);
```
and pass in here (and further below).